src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -226,29 +241,40 @@ static void ValidateNoReferenceToReference (LocalVariableStore locals, MethodDef
 		// reachable from it.
 		public virtual void InterproceduralScan (MethodBody methodBody)
 		{
-			var methodsInGroup = new ValueSet<MethodProxy> (methodBody.Method);
-
-			// Optimization to prevent multiple scans of a method.
-			// Eventually we will need to allow re-scanning in some cases, for example
-			// when we discover new inputs to a method. But we aren't doing dataflow across
-			// lambdas and local functions yet, so no need for now.
-			HashSet<MethodDefinition> scannedMethods = new HashSet<MethodDefinition> ();
+			var methodGroupLattice = new ValueSetLattice<MethodProxy> ();
+			// Note that the default value of a hoisted local will be MultiValueLattice.Top, not UnknownValue.Instance.
+			// This ensures that there are no warnings for the "unassigned state" of a parameter.
+			// Definite assignment should ensure that there is no way for this to be an analysis hole.
+			// When getting a hoisted local value, we also assert that the value is not Top.
+			var hoistedLocalsLattice = new DictionaryLattice<HoistedLocalKey, MultiValue, ValueSetLattice<SingleValue>> (
+				MultiValueLattice
+			);
+			var interproceduralStateLattice = new InterproceduralStateLattice (methodGroupLattice, hoistedLocalsLattice);
```

Comment:
It looks like `methodGroupLattice` and `hoistedLocalsLattice` are only used in the `InterproceduralStateLattice` ctor, maybe we should inline them to make it clear they're not used anywhere else.